### ISSUE TITLE
subscribeEvents can still raise on a closing connection

### ISSUE THREAD

**Issue by balloob** [home-assistant/home-assistant-js-websocket#65]
Since `subscribeEvents` returns  `info.unsubscribe` and not the promise there is no way to catch a rejection from the promise inside. This rejection occurs when the promise in `subscribeEvents` is called and has not been resolved by the time the connection is closed. The `_handleClose` is where the reject is thrown.

https://github.com/home-assistant/home-assistant-js-websocket/blob/2bc4cac3491e2c6aaddf2b1602af9495536eca7c/lib/connection.ts#L169-L200

A simple solution is to just use a `.catch(() => {})` on the promise. The same reasoning can be used here as you did for the `SendMessage` try/catch block in the promise
```
// Happens when the websocket is already closing.
// Don't have to handle the error, reconnect logic will pick it up.
```

_Originally posted by @zachowj in https://github.com/home-assistant/home-assistant-js-websocket/issues/59#issuecomment-434438462_

**Comment by balloob** [comment]
The current API of `subscribeEvents` is that it returns a promise that resolves the moment the subscription is in place. A catch is not the right solution as it means that we would resolve this promise before the subscription is in place.

**Comment by balloob** [comment]
I opened #66 which should fix the problem. Let me know if that works for you.

**Comment by zachowj** [comment]
Yes, this fixes the issue.

Thank you

**Comment by balloob** [comment]
Released as 3.1.6
